src/main/java/IP/duke/main/Ui.java

```diff
@@ -0,0 +1,160 @@
+package IP.duke.main;
+
+import IP.duke.task.Task;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents user interface abstraction.
+ * 
+ * @author Gordon Yit
+ * @Version CS2103T
+ */
+public class Ui {
+    private final String LOGO = " ____        _        \n" 
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
+    private String c = "~";
+    private String firstLine = " Hello! I'm Duke.Duke ";
+    private String secondLine = " What do you wanna do today? ";
+    private int bufferLength = 5;
+    private int width = Integer.max(firstLine.length(), secondLine.length()) + bufferLength;
+    private String buffer1 = c.repeat((width - firstLine.length()) / 2);
+    private String buffer2 = c.repeat((width - secondLine.length()) / 2);
+    private String welcomeMessage = buffer1 + firstLine + buffer1 + "\n" + buffer2 + secondLine + buffer2 + "\n";
+    private final String GREETING = "Hello from\n" + LOGO + welcomeMessage;
```

Comment:
Can possibly group constants at the start of the file instead of after normal variables. I think checkstyle checks for this so once you add checkstyle it should detect!